backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneTextMatchPredicateBuilder.java

```diff
@@ -47,17 +59,18 @@ public void fuzzy(int maxEditDistance, int exactPrefixLength) {
 
 	@Override
 	public void analyzer(String analyzerName) {
-		// TODO must be implemented
+		this.analyzerName = analyzerName;
 	}
 
 	@Override
 	protected Query doBuild(LuceneSearchPredicateContext context) {
 		if ( queryBuilder != null ) {
 			QueryBuilder effectiveQueryBuilder;
 			if ( maxEditDistance != null ) {
-				effectiveQueryBuilder = new FuzzyQueryBuilder( queryBuilder.getAnalyzer(), maxEditDistance, prefixLength );
+				effectiveQueryBuilder = new FuzzyQueryBuilder( getAnalyzer(), maxEditDistance, prefixLength );
 			}
 			else {
+				queryBuilder.setAnalyzer( getAnalyzer() );
```

Comment:
You're right. I didn't get that the instance was shared. Using the analyzer as a field instead of the query builder, I think I've solved also the issue you pointed just above this one.